on_device_training/web/README.md

```diff
@@ -0,0 +1,53 @@
+# End-to-end MNIST training demo
+
+This folder contains a demo based on [juharris's project](https://juharris.github.io/train-pytorch-in-js/) that launches a simple model that will train on the [MNIST dataset](http://yann.lecun.com/exdb/mnist/) in your browser.
+
+A live preview of this demo is available [here](https://carzh.github.io/onnxruntime-training-examples/).
+
+## Run instructions
+### 1. Prepare offline artifacts.
+On-device-training requires an offline step where 4 training artifacts are generated. The instructions for generating the required artifacts can be found in [`./offline-step/README.md`](offline-step/README.md). 
+
+Once the artifacts have been generated, copy them to [`./web-bundler/public`](web-bundler/public).
+```
+cp offline-step/*.onnx web-bundler/public
+cp offline-step/checkpoint web-bundler/public
+```
+
+### 2. Prepare data.
+This demo uses the MNIST dataset, which can be downloaded from [here](http://yann.lecun.com/exdb/mnist/). 
+
+Download the following files:
+* train-labels-idx1-ubyte.gz
+* train-images-idx3-ubyte.gz
+* t10k-labels-idx1-ubyte.gz
+* t10k-images-idx3-ubyte.gz
+
+You can use the gzip utility to unzip them.
+```
+gunzip train-labels-idx1-ubyte.gz train-images-idx3-ubyte.gz t10k-labels-idx1-ubyte.gz t10k-images-idx3-ubyte.gz
+```
+
+Once unzipped, they should be copied to the [`web-bundler/public/data`](web-bundler/public/data) folder.
+
+### 3. Run the demo.
+```
+cd web-bundler
+npm install
+npm run start
+```
+
+## onnxruntime-web/training npm package usage
```

Comment:
For web novices like me, could you also give an overview of how the *.html, *.txs files in this demo are used? Or maybe this information is already known to users and does not need to be mentioned here?

Reply:
would appreciate any feedback on detail level & clarity on the Demo components overview pushed to the README!!